src/common/actions/repository.js

```diff
@@ -16,104 +14,80 @@ export function toggleRepositorySelection(id) {
   };
 }
 
-export const REPO_ADD = 'REPO_ADD';
-export const REPO_SUCCESS = 'REPO_SUCCESS';
-export const REPO_FAILURE = 'REPO_FAILURE';
+export const REPO_ADD_REQUEST = 'REPO_ADD_REQUEST';
+export const REPO_ADD_SUCCESS = 'REPO_ADD_SUCCESS';
+export const REPO_ADD_FAILURE = 'REPO_ADD_FAILURE';
 export const REPO_RESET = 'REPO_RESET';
 
-export function addRepos(repositories, owner) {
-  return (dispatch) => {
-    const promises = repositories.map(
-      (repository) => {
-        return dispatch(addRepo(repository));
-      }
-    );
-    return Promise.all(promises).then(() => {
+export function addRepos(repos, owner) {
+  return async (dispatch) => {
+    try {
+      await Promise.all(repos.map(async (repo) => {
+        await dispatch(createWebhook(repo));
```

Comment:
Extracted dispatch of `createWebhook` from `addRepo` so that each can be rewritten to use the CALL_API middleware.